Редактирование Feature файла

phazmik opened the pull request:
Исправление идентификатора, описания и функционала существующего неимплементируемого …теста
Добавление нового неимплементируемого теста

phazmik committed:
Исправление описания и функционала существующего неимплементируемого теста
Добавление описания и функционала нового неимплементируемого теста

phazmik committed:
Внесены измениния по треду

phazmik committed:
Изменения по треду
Внесены изменения, касающиеся сортировки степов в feature файле
Названия папок и файлов скорректированы
Создан родительский файл для наследования объектов, находящихся на всех страницах

phazmik committed:
Внесение изменений в фиче и степах по треду
Изменение принципа работы некоторых шагов в тест-кейсах
Добавление вспомогательных шагов

phazmik committed:
Изменения фичи

Унификация тестов, проверяющих отображение предуепреждений

chinochee commented on PageObjectModels/RecoveryPage.cs:
```diff
@@ -1,35 +1,20 @@
         /// <summary>
         /// Элемент кнопки Назад
         /// </summary>
         public IWebElement backButtonElement => _driver.FindElement(By.XPath("//button [1]"));
```
Наименование всех свойств в моделях страниц должны начинаться с заглавной буквы

phazmik replied:
Исправлено

chinochee commented on PageObjectModels/RecoveryPage.cs:
```diff
@@ -1,35 +1,20 @@
+namespace LifePayTests.PageObjectModels
 {
-    public class PageRecoveryObjectModel
+    public class RecoveryPage : CommonPageObjects
```
Добавить описание к моделям страниц

phazmik replied:
Исправлено

chinochee commented on StepDefinitions/AuthPageSteps.cs:
```diff
@@ -11,45 +12,16 @@ namespace LifePayTests.StepDefinitions
         {
             var driver = (IWebDriver)_scenarioContext["driver"];
-            var _pageObjectModel = new PageSignInObjectModel(driver);
+            var _pageObjectModel = new SignInPage(driver);
```
Наименование переменных изменять в соответствии с классом: в данном случае _pageObjectModel => signInPage
Нижнее подчеркивание вначале наименования переменной нужно для обозначения, что она приватная. Здесь у тебя переменная _pageObjectModel **не приватная**

phazmik replied:
Исправлено

chinochee replied:
Перечитай первое предложение
Название переменной должно совпадать с названием класса

phazmik replied:
Проглядел, исправил

chinochee commented on StepDefinitions/AuthPageSteps.cs:
```diff
@@ -61,129 +33,99 @@ public void WhenПроисходитНажатиеНаСсылку(string linkNa
-                Assert.AreEqual(toolsForTests._pass, currentPass);
-                driver.Close();
+                Assert.AreEqual(link, currentURL);
             }
-            catch 
+            catch
             {
-                Console.WriteLine("Fillment of fields is wrong");
-            } 
+                Debug.WriteLine("URL is wrong");
+            }
```
Убрать трай-кетч и логирование в дебаг консоль

phazmik replied:
Исправлено

chinochee commented on Features/AuthPage.feature:
```diff
@@ -53,34 +53,54 @@
+@id_incorrest_password
+Примеры:
+| Пароль | Предупреждение                         |
+| @1234  | Значение должно быть не менее 6 знаков |
```
Где проверка "не более"?

chinochee replied:
если такая текстовка вообще есть. Мб там лимит на ввод стоит. Так что могу ошибаться. Проверь это. Если есть текстовка Не более, то напиши проверку

phazmik replied:
Стоит лимит на ввод, поэтому проверку на "не более" не делал

chinochee commented on Features/AuthPage.feature:
```diff
@@ -53,34 +53,54 @@
+Сценарий: Отображение предупреждения поля Номер телефона при введении не зарегистрированного номера телефона
 	Допустим Открыта страница авторизации LifePay 
-	И незарегистрированный номер известен
-	Когда поле ввода Номер телефона заполнено
-	Тогда отображается сообщение Номер не зарегистрирован
+	Когда поле ввода <имя Поля> заполнено значением '<Значение>'
+	Тогда отображается предупреждение '<Предупреждение>' в поле Номер телефона 
+
+@id_unreg_phone
+Примеры:
+| Тип значения | Имя поля     | Значение         | Предупреждение           |
+|              | Номер телефона | +7(911)111-11-12 | Номер не зарегистрирован |
+
+@id_incorrect_phone
+Примеры:
+| Номер телефона  | Предупреждение                             |
+| +7(911)111-11-1 | Введите телефон в формате +7(911)111-11-11 |
```
Ты дописал автотесты? Значения, передаваемые в степы сценария должны быть переданы все. У тебя не должно быть 4 или 2 свойства для одного сценария. Нужно либо 4, либо 2 свойства передавать. Если потребуется, то разделяй сценарий на 2 сценария

phazmik replied:
Это было не доделано. Берётся 3 свойства, имплементация в работе.

chinochee commented on Features/AuthPage.feature:
```diff
@@ -53,34 +53,54 @@
+Сценарий: Успешная авторизация
+	Допустим Открыта страница авторизации LifePay
+	Когда поле ввода Номер телефона заполнено значением 
+	И поле ввода Пароль заполнено значением
+	И нажата кнопка Войти в личный кабинет
+	Тогда происходит редирект в личный кабинет	
+
+@successful_reg_no_inplimentation
 Сценарий: Успешная регистрация
 	Допустим Открыта страница авторизации LifePay
-	И данные регистрации известны 
-	Когда поле ввода Номер телефона заполнено
+	Когда поле ввода Номер телефона заполнено значением
 	И поле ввода Код из SMS заполнено
 	И поле ввода Ваше имя и фамилия заполнено
-	И поле ввода Пароль заполнено
+	И поле ввода Пароль заполнено значением
 	И нажата кнопка Создать личный кабинет
 	Тогда происходит редирект в личный кабинет
```
2 сценария авторизации? Или авторизация и регистрация?
Вообще проверки на регистрацию делать не нужно. И давай на авторизацию тоже не делать

phazmik replied:
Тест-кейсы по успешной авторизации и регистрации без имплементации

chinochee commented on Features/AuthPage.feature:
```diff
@@ -31,18 +31,18 @@
 @successful_auth_field_fillment
```
Добавь уникальные идентификаторы для сценариев: это нужно для того, чтобы можно было проще искать по проекту. Формат id придумай сам - главное, что он был одинаковый и не громоздкий

phazmik replied:
Исправлено

phazmik committed:
Изменения в фиче и степах
Свёртывание некоторых степов в унифицированый с подстановкой необходимых значений

phazmik committed:
Обозначение предстоящих изменений в фиче
Обозначение степов для изменений в фиче

phazmik committed:
Исправления по тредам

phazmik committed:
Внесены изменения по тредам

chinochee commented on PageObjectModels/SignUpPage.cs:
```diff
@@ -1,102 +1,80 @@
 {
-    public class PageSignUpObjectModel
+    /// <summary>
+    /// Элементы страницы Регистрации
```
Модель страница Регистрация
или
Страница Регистрация
или 
Страница регистрации

Такое же замечание по остальным моделям страниц

chinochee commented on StepDefinitions/AuthPageSteps.cs:
```diff
@@ -1,189 +1,140 @@
                 case "Восстановить пароль":
-                    _pageObjectModel.linkRecoveryPasswordElement.Click();
-                    break;
+                    {
```
Контекст здесь не нужен

phazmik replied:
Тогда откуда брать драйвер для класса signInPage?

chinochee replied:
у тебя в мейне нет контекста и все работало
![image](https://github.com/phazmik/LifePayTests/assets/66793241/a5a30f4d-65b6-43f6-8896-3e3e3a9464e4)
Драйвер ты выше берешь. Не монимаю, почему ты задаешь вопрос именно продрайвер

chinochee commented on StepDefinitions/AuthPageSteps.cs:
```diff
@@ -1,189 +1,140 @@
 using NUnit.Framework;
 using OpenQA.Selenium;
+using PageObjectModels;
+using Support;
```
Провесь библиотеки. Неиспользуемые библиотеки удали

phazmik replied:
все указанные библиотеки используются, неиспользуемые не держу

phazmik committed:
Внесены изменения по тредам

phazmik committed:
Исправления по треду и в фиче
